enderio-base/src/main/java/com/enderio/base/data/recipe/standard/MaterialRecipes.java

```diff
@@ -29,5 +41,231 @@ protected void buildCraftingRecipes(@Nonnull Consumer<FinishedRecipe> recipeCons
+        ingotToNuggets(recipeConsumer, EIOItems.SOULARIUM_NUGGET.get(), EIOItems.SOULARIUM_INGOT.get());
+        blockToIngots(recipeConsumer, EIOItems.END_STEEL_INGOT.get(),EIOBlocks.END_STEEL_BLOCK.get());
+        ingotToNuggets(recipeConsumer, EIOItems.END_STEEL_NUGGET.get(), EIOItems.END_STEEL_INGOT.get());
+        blockToIngots(recipeConsumer, EIOItems.CONSTRUCTION_ALLOY_INGOT.get(),EIOBlocks.CONSTRUCTION_ALLOY_BLOCK.get());
+        ingotToNuggets(recipeConsumer, EIOItems.CONSTRUCTION_ALLOY_NUGGET.get(), EIOItems.CONSTRUCTION_ALLOY_INGOT.get());
+        
+        ShapedRecipeBuilder.shaped(EIOItems.CONDUIT_BINDER_COMPOSITE.get())
+            .pattern("GCG")
+            .pattern("SGS")
+            .pattern("GCG")
+            .define('G', Tags.Items.GRAVEL)
+            .define('S', Tags.Items.SAND)
+            .define('C', Items.CLAY_BALL)
+            .unlockedBy("has_ingredient_gravel", InventoryChangeTrigger.TriggerInstance.hasItems(Items.GRAVEL))
+            .unlockedBy("has_ingredient_sand", InventoryChangeTrigger.TriggerInstance.hasItems(Items.SAND))
+            .unlockedBy("has_ingredient_clay", InventoryChangeTrigger.TriggerInstance.hasItems(Items.CLAY_BALL))
+            .save(recipeConsumer);
+        
+        ShapedRecipeBuilder.shaped(EIOItems.PULSATING_CRYSTAL.get())
+            .pattern("PPP")
+            .pattern("PDP")
+            .pattern("PPP")
+            .define('P', EIOItems.PULSATING_IRON_NUGGET.get())
+            .define('D', Tags.Items.GEMS_DIAMOND)
+            .unlockedBy("has_ingredien", InventoryChangeTrigger.TriggerInstance.hasItems(EIOItems.PULSATING_IRON_NUGGET.get()))
+            .save(recipeConsumer);
+        
+        ShapedRecipeBuilder.shaped(EIOItems.VIBRANT_CRYSTAL.get())
+            .pattern("PPP")
+            .pattern("PDP")
+            .pattern("PPP")
+            .define('P', EIOItems.VIBRANT_ALLOY_NUGGET.get())
+            .define('D', Tags.Items.GEMS_EMERALD)
+            .unlockedBy("has_ingredien", InventoryChangeTrigger.TriggerInstance.hasItems(EIOItems.VIBRANT_ALLOY_NUGGET.get()))
+            .save(recipeConsumer);
+        
+        ShapedRecipeBuilder.shaped(EIOItems.DYE_INDUSTRIAL_BLEND.get())
+            .pattern("LQG")
+            .pattern("QBQ")
+            .pattern("GQL")
+            .define('L', EIOTags.Items.DUSTS_LAPIS)
+            .define('Q', EIOTags.Items.DUSTS_QUARTZ)
+            .define('B', EIOItems.DYE_BLACK.get())
+            .define('G', EIOItems.DYE_GREEN.get())
+            .unlockedBy("has_ingredient_black", InventoryChangeTrigger.TriggerInstance.hasItems(EIOItems.DYE_BLACK.get()))
+            .unlockedBy("has_ingredient_green", InventoryChangeTrigger.TriggerInstance.hasItems(EIOItems.DYE_GREEN.get()))
+            .unlockedBy("has_ingredient_lapis", InventoryChangeTrigger.TriggerInstance.hasItems(ItemPredicate.Builder.item().of(EIOTags.Items.DUSTS_LAPIS).build()))
```

Comment:
Is checking tags really this long? Surely there's a neater way, no? If not no worries, just though there'd be a way of doing this in less code haha

Reply:
Well Vanilla does it the same way. I also find it stupid, but I don't see another way.